misk/src/test/kotlin/misk/digester/WindowerTest.kt

```diff
@@ -0,0 +1,215 @@
+package misk.digester
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.util.Calendar
+import kotlin.test.assertEquals
+class WindowerTest {
+  @Test
+  fun testWindowerInitialBoundaries() {
+    assertThat(Windower(60, 1).startSecs).isEqualTo(mutableListOf(0))
+    assertThat(Windower(30, 1).startSecs).isEqualTo(mutableListOf(0, 30))
+    assertThat(Windower(60, 2).startSecs).isEqualTo(mutableListOf(0, 30))
+    assertThat(Windower(60, 3).startSecs).isEqualTo(mutableListOf(0, 20, 40))
+  }
+  private data class WindowerTestClass(
```

Comment:
Can you look into https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests? Seems like the canonical JUnit way to do table tests.

Reply:
Using parameterized tests would not make the testsuite more clear or readable. I don't think there is an advantage to using them in this situation as a custom data class would still be needed.